# Generated by Django 4.2.16 on 2024-11-12 23:40

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('accounts', '0002_user_role'),
    ]

    operations = [
        migrations.AddField(
            model_name='user',
            name='phone_code',
            field=models.CharField(default=1, max_length=3),
            preserve_default=False,
        ),
        migrations.AlterField(
            model_name='user',
            name='phone_number',
            field=models.CharField(max_length=10),
        ),
        migrations.AlterField(
            model_name='user',
            name='role',
            field=models.IntegerField(choices=[(1, 'Admin'), (2, 'Lessee'), (3, 'Lessor')], null=True),
        ),
        migrations.AddConstraint(
            model_name='user',
            constraint=models.UniqueConstraint(fields=('phone_number', 'phone_code'), name='ux_phone', violation_error_message='User with given phone number already exists.'),
        ),
    ]
